src/components/Banner/Banner.tsx

```diff
@@ -2,108 +2,162 @@ import clsx from 'clsx';
 import React, { ReactNode, useState } from 'react';
 import styles from './Banner.module.css';
 import Button from '../Button';
-import Icon, { IconName } from '../Icon';
+import Icon from '../Icon';
 
 export interface Props {
   /**
-   * The child node(s) contains the banner message. Note: the banner message is displayed inside a TextPassage, so children can contain raw HTML
+   * A button or link that's placed in the banner separately from the main content.
    */
-  children?: ReactNode;
+  action?: React.ReactNode;
   /**
    * CSS class names that can be appended to the component.
    */
   className?: string;
-  /**
-   * Close button text
-   * 1) Used to change the text of the icon button with various languages
-   */
-  closeButtonText?: string;
   /**
    * Toggles the ability to dismiss the banner via an close button in the top right of the banner
    */
-  dismissible?: boolean;
-  /**
-   * Name of banner SVG icon. Available options are: banner, error, warning-2, success
-   */
-  iconName?: IconName;
+  dismissable?: boolean;
   /**
-   * The title attribute for the SVG icon in the banner
+   * Optional callback to be triggered when the banner is dismissed
    */
-  iconTitle?: string;
+  onDismiss?: () => void;
   /**
    * Controls the layout of the banner
    * - **vertical** renders the banner content center aligned and stacked
+   *
+   * Vertical banners are used in narrow areas, like sidebars
    */
   orientation?: 'vertical';
   /**
-   * Optional heading text for the banner that appears above banner body message
+   * The child node(s) contains the banner message
    */
-  title?: string;
+  text?: ReactNode;
```

Comment:
I worry that using `text` here contradicts our own guidance to "default to `text` for short strings of text". I know we haven't revisited this in a while 😅 , but could we consider passing
- `title` prop to replace `BannerTitle`
- `titleAs` or `headingAs` prop to toggle heading element if needed
-  `description` to replace `BannerDescription`

? right now the Banner API is kind of a snowflake, so trying to balance that with making sure people use the `action` prop

Reply:
Ah, thanks! I also felt like `text` didn't quite fit but also thought it fit better than `textContent`. I'm just wondering if the types for these props should be `string` or whether we want to allow them to pass in react nodes in the rare case that they need to override the text styles for some reason 🤔

Reply:
Actually the `description` definitely needs to be a `React.Node` so they can embed links. I could see the `title` needing test ids or other attributes, so I think `React.Node` is the way to go in this case.

Reply:
sounds fine to me 👍